src/components/negotiations-panel.tsx

```diff
@@ -0,0 +1,256 @@
+  NegotiationStatus,
+  ResolveState
+} from '../negotiation-status-manager';
+
+import {
+  RobotTrajectoryManager,
+  NegotiationTrajectoryResponse
+} from '../robot-trajectory-manager';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    height: 240,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+  finished: {
+    backgroundColor: "lightgreen"
+  },
+  rejected: {
+    backgroundColor: theme.palette.warning.main
+  },
+  forfeited: {
+    backgroundColor: theme.palette.warning.main
+  },
+  defunct: {
+    backgroundColor: theme.palette.error.main
+  },
+  unresolved: {
+    backgroundColor: theme.palette.error.main
+  },
+  ongoing: {
+    backgroundColor: "yellow"
+  }
+}));
+
+interface Parameter
+{
+  conflict_version : number;
+  sequence : number[];
+};
+
+export interface NegotiationsPanelProps {
+  conflicts : Readonly<Record<string, NegotiationConflict>>;
+  spotlight?: Readonly<SpotlightValue<string>>;
+  trajManager?: Readonly<RobotTrajectoryManager>;
+  negotiationTrajStore : Record<string, NegotiationTrajectoryResponse>;
+}
+
+export default function NegotiationsPanel(props: NegotiationsPanelProps): JSX.Element {
+  const { conflicts, spotlight, trajManager, negotiationTrajStore } = props;
+
+  React.useEffect(() => {
+    if (!spotlight) {
+      return;
+    }
+    // TODO: spotlight
+  }, [spotlight]);
+  
+  const classes = useStyles();
+
+  /**  Utility conversion functions **/
+  const determineStatusText = (status : NegotiationStatus,
+    parent_resolved : ResolveState): string => {
+    if (status.forfeited)
+      return "  [FORFEITED]";
+    else if (status.rejected)
+      return "  [REJECTED]";
+    else if (status.defunct)
+      return "  [DEFUNCT]";
+    else if (parent_resolved === ResolveState.RESOLVED)
+      return "  [FINISHED]";
+    else
+      return "  [ONGOING]";
+  };
+  const determineStyle = (status : NegotiationStatus, 
+    parent_resolved : ResolveState): string => {
+    if (status.forfeited)
+      return classes.forfeited;
+    else if (status.rejected)
+      return classes.rejected;
+    else if (status.defunct)
+      return classes.defunct;
+    else if (parent_resolved === ResolveState.RESOLVED)
+      return classes.finished;
+    else
+      return classes.ongoing;
+  };
+
+  // keep track of parameters so we can send them as requests
+  let nodeidToParameters = new Map<string, Parameter>();
+
+  /** Render Negotiation panel contents **/
+  const renderNegotiations = (version : string, conflict : NegotiationConflict) => {
+    let conflictLabel = "Conflict #" + version + ", Participants: ";
+    let i = 0;
+    for (const name of Object.values(conflict.participantIdsToNames))
+    {
+      conflictLabel += name;
+
+      if (i !== (Object.keys(conflict.participantIdsToNames).length - 1))
+        conflictLabel += ", ";
+      ++i;
+    }
+    let conflictStyle = classes.ongoing;
+    if (conflict.resolved === ResolveState.RESOLVED)
+      conflictStyle = classes.finished;
```

Comment:
What do you think about creating a function `getConflictStyle` as you did with `determineStyle`?

```
const getConflictStyle =() =>{
  if (conflict.resolved === ResolveState.RESOLVED)
      return classes.finished;
    else if (conflict.resolved === ResolveState.FAILED)
      return classes.unresolved;

    return classes.ongoing
}
```

const conflictStyle = getConflictStyle()

Reply:
`determineStyle` and its variants are used repeatedly multiple times hence the abstraction. In this case it's only used in a single place. We could make more abstraction but I'd generally stay away from that since there's more jumping around for definitions as you read the code.

If it still irks you and the team, let me know and I'll change it.

Reply:
Cool, thanks for the feedback.

Actually, AFAIK there is nothing definite regarding this topic in this project, but it is good that you raised it. I guess as long as the code is understood and testable, it's fine by me. So feel free to leave it as is. Thank you.